### PR TITLE
LibWeb: Add start of HTML Tokenizer in Swift

### PR DESCRIPTION
**Pull request by ADKaster**
Currently it's just a Token class.

### CODE REVIEW

**Comment by awesomekling** [comment]
Typo: tokenzier -> tokenizer

More importantly, macOS CI seems upset 🤔

File: Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        var namespace_: String?
+        var value: String
+        var nameStartPosition: Position
+        var nameEndPosition: Position
+        var valueStartPosition: Position
+        var valueEndPosition: Position
+    }
+
+    public enum TokenType {
+        case Invalid
```
Comment (dzfrias) [inline Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift:26]: Are we doing upper camel case for enum variants? Foundation and the docs seem to use lower camel case.

**Comment by ADKaster** [comment]
whoops, pushed some wip stuff to this branch. ... feels like a tomorrow problem though

File: Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift (hunk 1 above)
Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift:26]: wait, really? ew. why would I want my variants to start with a lower case ;-;. sigh. I'll follow the 'official' style though.

Comment (alimpfard) [reply Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift:26]: Err...why?
we ignored plenty of C++ style rules, if it looks better, it looks better.

Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift:26]: IMO, if these are going to be visible from C++ land, keep the capitalisation consistent with that.

Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift:26]: eh, all right. I'll leave this as is until we get more swift files in the repo.

**Pull request merged**